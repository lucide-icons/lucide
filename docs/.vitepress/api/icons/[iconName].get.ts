import { eventHandler, getQuery, setResponseHeader, createError } from 'h3'
import iconNodes from '../../data/iconNodes'
import createLucideIcon from 'lucide-react/src/createLucideIcon'
import { renderToString } from 'react-dom/server'
import { createElement } from 'react'

export default eventHandler((event) => {
  const { params } = event.context

  const iconNode = iconNodes[params.iconName]

  if (iconNode == null) {
    const error = createError({
      statusCode: 404,
      message: `Icon "${params.iconName}" not found`,
    })

    return sendError(event, error)
  }

  const width = getQuery(event).width || undefined
  const height = getQuery(event).height || undefined
  const color = getQuery(event).color || undefined
  const strokeWidth = getQuery(event).strokeWidth || undefined

  const LucideIcon = createLucideIcon(params.iconName, iconNode)

  const svg = Buffer.from(
    renderToString(
      createElement(LucideIcon, {
        width,
        height,
        color: color ? `#${color}` : undefined,
        strokeWidth,
      }
    ))
  ).toString('utf8');

  defaultContentType(event, 'image/svg+xml')
  setResponseHeader(event, 'Cache-Control', 'public,max-age=31536000')

  return svg

})
