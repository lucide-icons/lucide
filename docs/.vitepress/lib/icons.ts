import fs from 'fs';
import path from 'path';
import { IconNodeWithKeys } from '../theme/types';
import iconNodes from '../data/iconNodes';
import releaseMeta from '../data/releaseMetaData.json';

const DATE_OF_FORK = '2020-06-08T16:39:52+0100';

const directory = path.join(process.cwd(), '../icons');

export interface GetDataOptions {
  withChildKeys?: boolean;
}

export async function getData(name: string) {
  const jsonPath = path.join(directory, `${name}.json`);
  const jsonContent = fs.readFileSync(jsonPath, 'utf8');
  const { tags, categories, contributors } = JSON.parse(jsonContent);

  const iconNode = iconNodes[name];

  const releaseData = releaseMeta?.[name] ?? {
    createdRelease: {
      version: '0.0.0',
      date: DATE_OF_FORK,
    },
    changedRelease: {
      version: '0.0.0',
      date: DATE_OF_FORK,
    },
  };

  return {
    name,
    tags,
    categories,
    iconNode,
    contributors,
    ...releaseData,
  };
}

export async function getAllData(): Promise<{ name: string; iconNode: IconNodeWithKeys }[]> {
  const names = Object.keys(iconNodes);

  return Promise.all(names.map((name) => getData(name)));
}
