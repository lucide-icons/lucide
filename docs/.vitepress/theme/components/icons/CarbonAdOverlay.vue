<script setup lang="ts">
import { useData } from 'vitepress';
import { useSessionStorage } from '@vueuse/core';
import IconButton from '../base/IconButton.vue';
import VPDocAsideCarbonAds from 'vitepress/dist/client/theme-default/components/VPDocAsideCarbonAds.vue'
import { x } from '../../../data/iconNodes'
import createLucideIcon from 'lucide-vue-next/src/createLucideIcon';

const { theme } = useData()
const showAd = useSessionStorage('show-carbon-ads',true)

defineProps<{
  drawerOpen: boolean
}>()

const CloseIcon = createLucideIcon('Close', x)
</script>

<template>
  <div
    :class="{
      'drawer-open': drawerOpen,
      'hide-ad': !showAd
    }"
    class="floating-ad"
    v-if="theme.carbonAds"
  >
    <IconButton @click="showAd = false" class="hide-button">
      <component :is="CloseIcon" :size="20" absoluteStrokeWidth />
    </IconButton>
    <VPDocAsideCarbonAds
      :carbon-ads="theme.carbonAds"
    />
  </div>
</template>

<style scoped>
.floating-ad {
  display: none;
  position: fixed;
  bottom: 32px;
  width: 224px;
  right: 32px;
  transition: opacity 0.5s, transform 0.25s ease;
}

.floating-ad.drawer-open {
  transform: translateY(-288px);
}

.floating-ad.hide-ad {
  transform: translateX(224px);
  opacity: 0;
}

.floating-ad.drawer-open.hide-ad {
  transform: translateY(-288px) translateX(224px);
}

.floating-ad.drawer-open, .floating-ad.hide-ad {
  transition: opacity 0.25s, transform 0.5s cubic-bezier(0.19, 1, 0.22, 1);
}

@media (min-width: 1280px) {
  .floating-ad {
    display: block;
  }
}

@media (min-width: 1440px) {
  .floating-ad {
    right: calc(((100% - (var(--vp-layout-max-width) - var(--vp-sidebar-width))) - 272px) / 2);
  }
}

.hide-button {
  padding: 4px;
  position: absolute;
  top: 8px;
  right: 8px;
  background-color: transparent;
}
</style>
