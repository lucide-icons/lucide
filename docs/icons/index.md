---
layout: page
outline: 2
outlineTitle: Categories
sidebar: true
---

<script setup>
import { data } from './icons.data.ts'
import IconsOverview from '../.vitepress/theme/components/icons/IconsOverview.vue'
import PageContainer from '../.vitepress/theme/components/PageContainer.vue'
</script>

<div class="VPDoc content">
  <PageContainer>
    <IconsOverview :icons="data.icons" />
  </PageContainer>
</div>
