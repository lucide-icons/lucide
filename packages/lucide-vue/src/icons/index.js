/*
 Icons exports.

 Will be generated
*/
