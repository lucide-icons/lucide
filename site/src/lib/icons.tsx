import fs from "fs";
import path from "path";
import tags from '../../../tags.json';

const directory = path.join(process.cwd(), "../icons");

export function getAllNames() {
  const fileNames = fs.readdirSync(directory);

  return fileNames.map((fileName) => {
    return fileName.replace(/\.svg$/, "");
  });
}

export function getData(name) {
  const fullPath = path.join(directory, `${name}.svg`);
  const fileContents = fs.readFileSync(fullPath, "utf8");

  return {
    name,
    tags: tags[name] || [],
    src: fileContents,
  };
}

export function getAllData() {
  const names = getAllNames();

  return names.map((name) => {
    return getData(name);
  });
}
