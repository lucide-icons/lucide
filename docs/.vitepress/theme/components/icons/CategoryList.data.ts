import { getAllData } from '../../../lib/icons';
import { getAllCategoryFiles, mapCategoryIconCount } from '../../../lib/categories';
import iconsMetaData from '../../../data/iconMetaData'


export default {
  async load() {
    let categories = getAllCategoryFiles()

    categories = mapCategoryIconCount(categories, Object.values(iconsMetaData))

    return {
      categories,
    }
  }
}
