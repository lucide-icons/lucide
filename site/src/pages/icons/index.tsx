import Layout from '../../components/Layout';
import IconOverview from '../../components/IconOverview';
import { getAllData } from '../../lib/icons';
import { getAllCategories } from 'src/lib/categories';
import IconDetailOverlay from 'src/components/IconDetailOverlay';

const IconsPage = ({ data, categories }) => {
  return (
    <Layout>
      <IconDetailOverlay />
      <IconOverview {...{ data, categories }} key="icon-overview" />
    </Layout>
  );
};

export default IconsPage;

export async function getStaticProps() {
  const data = await getAllData({ withChildKeys: true });
  const categories = await getAllCategories()

  return {
    props: {
      data,
      categories,
    },
  };
}
