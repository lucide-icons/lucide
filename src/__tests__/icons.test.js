/* eslint-env jest */
import icons from '../icons';

jest.mock('../../dist/icons.json', () => ({
  icon1:
    '<line x1="23" y1="1" x2="1" y2="23" /><line x1="1" y1="1" x2="23" y2="23" />',
  icon2: '<circle cx="12" cy="12" r="11" />',
}));

jest.mock('../tags.json', () => ({
  icon1: ['foo', 'bar', 'hello', 'world'],
}));

test('exports correct object', () => {
  expect(icons).toMatchSnapshot();
});
