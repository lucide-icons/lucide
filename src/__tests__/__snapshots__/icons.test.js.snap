// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports correct object 1`] = `
Object {
  "icon1": Icon {
    "attrs": Object {
      "class": "feather feather-icon1",
      "fill": "none",
      "height": 24,
      "stroke": "currentColor",
      "stroke-linecap": "round",
      "stroke-linejoin": "round",
      "stroke-width": 2,
      "viewBox": "0 0 24 24",
      "width": 24,
      "xmlns": "http://www.w3.org/2000/svg",
    },
    "contents": "<line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\" /><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\" />",
    "name": "icon1",
    "tags": Array [
      "foo",
      "bar",
      "hello",
      "world",
    ],
  },
  "icon2": Icon {
    "attrs": Object {
      "class": "feather feather-icon2",
      "fill": "none",
      "height": 24,
      "stroke": "currentColor",
      "stroke-linecap": "round",
      "stroke-linejoin": "round",
      "stroke-width": 2,
      "viewBox": "0 0 24 24",
      "width": 24,
      "xmlns": "http://www.w3.org/2000/svg",
    },
    "contents": "<circle cx=\\"12\\" cy=\\"12\\" r=\\"11\\" />",
    "name": "icon2",
    "tags": Array [],
  },
}
`;
